lxd/main_init.go

```diff
@@ -417,18 +421,56 @@ they otherwise would.
 				return err
 			}
 
+			defaultProfileExists := false
 			for _, p := range profiles {
 				if p.Name != "default" {
 					continue
 				}
 
+				defaultProfileExists = true
+
+				foundRootDiskDevice := false
+				for k, v := range p.Devices {
+					if v["path"] == "/" && v["source"] == "" {
+						foundRootDiskDevice = true
+
+						// Unconditionally overwrite
+						// because if the user ends up
+						// with a clean LXD but with a
+						// pool property key existing in
+						// the default profile it must
+						// be empty otherwise it would
+						// not have been possible to
+						// delete the storage pool in
+						// the first place.
```

Comment:
We don't stick to 80 cols in the LXD code, you can wrap that comment to be a bit more readable :)